docroot/profiles/custom/cgov_site/modules/custom/cgov_site_section/cgov_site_section.module

```diff
@@ -119,35 +164,178 @@ function cgov_site_section_taxonomy_term_update(EntityInterface $entity) {
-    $entity->get('computed_path')->value &&
-    $entity->original->get('parent')->target_id ===
-    $entity->get('parent')->target_id &&
-    $entity->original->get('field_pretty_url')->value ===
-    $entity->get('field_pretty_url')->value &&
-    $entity->original->get('langcode')->value ===
-    $entity->get('langcode')->value
-  ) {
-    return;
+  // Save the term children if this term update meets the criteria.
+  if (will_trigger_save_children($entity)) {
+    save_term_children($entity);
   }
-  $term_storage = \Drupal::service('entity_type.manager')->getStorage("taxonomy_term");
+  // Determine if this term update meets the criteria for tag invalidation.
+  if (_will_trigger_cache_tag_invalidation($entity)) {
+    invalidate_section_cache_tag($entity);
+  }
+
+  // TODO: Figure out how to get pathauto to update all the aliases using
+  // pathauto. To do it here is A) too slow, and B) fraught with
+  // issue regarding workflow and the various entity types we would have to hit.
+}
 
+/**
+ * Invalidate the cache tag equal to the given Site Sections ID.
+ *
+ * @param \Drupal\Core\Entity\EntityInterface $site_section_entity
+ *   The Site Section to invalidate.
+ */
+function invalidate_section_cache_tag(EntityInterface $site_section_entity) {
+  $section_id = $site_section_entity->id();
+  // Get the sections ID and invalidate the cache tag.
+  if ($section_id) {
+    Cache::invalidateTags(['site_section:' . $section_id]);
+  }
+}
+
+/**
+ * Saves the children of the given Site Section taxonomy term.
+ *
+ * @param \Drupal\Core\Entity\EntityInterface $updated_section_entity
+ *   The site section entity with its pending value updates.
+ */
+function save_term_children(EntityInterface $updated_section_entity) {
+  $term_storage = \Drupal::service('entity_type.manager')
+    ->getStorage("taxonomy_term");
   // Update immediate child sections.
-  $subsections = $term_storage->loadChildren($entity->id(), 'cgov_site_sections');
+  $subsections = $term_storage->loadChildren($updated_section_entity->id(), 'cgov_site_sections');
 
-  // Resave the sections. This should trigger the presave above with will set
+  // Re-save the sections. This should trigger the presave above with will set
   // the computed_path to the new path.
   foreach ($subsections as $section) {
     $section->save();
   }
+}
 
-  // TODO: Figure out how to get pathauto to update all the aliases using
-  // pathauto. To do it here is A) too slow, and B) fraught with
-  // issue regarding workflow and the various entity types we would have to hit.
+/**
+ * Determines if a section should trigger a re-save of its children terms.
+ *
+ * @param \Drupal\Core\Entity\EntityInterface $updated_section_entity
+ *   The site section entity with its pending value updates.
+ *
+ * @return bool
+ *   A TRUE or FALSE as to whether to trigger a re-save.
+ */
+function will_trigger_save_children(EntityInterface $updated_section_entity) {
+
```

Comment:
Returns whether or not a site section  should update it children, based on if certain fields have changed.

Reply:
Looks good.